### PR TITLE
Code review of filter search feature 

### PR DESCRIPTION
**Pull request by ayuzhin**
# Problem Overview 
Code review of screen layout new search filtering feature.

# Proposed Solution
1. Added new button type `.blank`
1. Removed magic numbers when customizing UI
1. Replacing part of the alignment code with the use of extensions
1. Fixed project structure

**Added reviewers**
* Lolita
* Pavel

### CODE REVIEW

File: Documents/Documents/Code/Components/Style/ASCButtonStyle.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import UIKit
 
 enum ASCButtonStyleType: Int {
-    case `default`, action, bordered, gray, capsule
+    case `default`, action, blank, bordered, gray, capsule
```
Comment (ayuzhin) [inline Documents/Documents/Code/Components/Style/ASCButtonStyle.swift:12]: Add new style of button

File: Documents/Documents/Code/Components/Style/ASCTextStyle.swift [hunk 2]
```diff
@@ -57,24 +59,31 @@ enum ASCTextStyle {
         case .semibodyWhite:
             return (UIFont.preferredFont(forTextStyle: .body).with(weight: .semibold), .white, 1)
 
-        case .subhead:
+        case .headline:
```
Comment (ayuzhin) [inline Documents/Documents/Code/Components/Style/ASCTextStyle.swift:62]: Add new text style for user as category header in filter screen

File: Documents/Documents/Code/Components/Style/ASCTextStyle.swift [hunk 3]
```diff
@@ -20,10 +20,12 @@ enum ASCTextStyle {
...
-    case subheadLight
-    case subhead
-    case subheadWhite
-    case subheadBold
+    case headline
+
+    case subheadlineLight
+    case subheadline
+    case subheadlineWhite
+    case subheadlineBold
```
Comment (ayuzhin) [inline Documents/Documents/Code/Components/Style/ASCTextStyle.swift:28]: Just refactoring

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 4]
```diff
@@ -9,72 +9,92 @@
...
+    }(ASCButtonStyle())
+
+    // MARK: - Lifecycle Methods
 
     override func viewDidLoad() {
         super.viewDidLoad()
         view.backgroundColor = Asset.Colors.tableCategoryBackground.color
         setupNavigationBar()
-        showResultButtonConstraints()
         setupCollectionView()
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:57]: Change an order to set button bring to front

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 5]
```diff
@@ -9,72 +9,92 @@
...
-
-        ])
+        showResultsButton.anchor(
+            left: view.leftAnchor,
+            bottom: view.safeAreaLayoutGuide.bottomAnchor,
+            right: view.rightAnchor,
+            leftConstant: 16,
+            bottomConstant: 10,
+            rightConstant: 16,
+            heightConstant: 52
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:78]: Use own extensions

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 6]
```diff
@@ -9,72 +9,92 @@
...
     func setupNavigationBar() {
-        let navigationTitle = UILabel()
-        navigationTitle.font = UIFont.systemFont(ofSize: 17, weight: .semibold)
-        navigationTitle.text = NSLocalizedString("Filters", comment: "")
-        navigationItem.titleView = navigationTitle
-
-        let rightBarButton = UIBarButtonItem(title: NSLocalizedString("Reset", comment: ""), style: UIBarButtonItem.Style.plain, target: self, action: #selector(resetBarButtonItemTapped))
-        let leftBarButton = UIBarButtonItem(title: NSLocalizedString("Cancel", comment: ""), style: UIBarButtonItem.Style.plain, target: self, action: #selector(cancelBarButtonItemTapped))
+        navigationItem.largeTitleDisplayMode = .never
+        navigationItem.title = NSLocalizedString("Filters", comment: "")
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:84]: Use system style

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 7]
```diff
@@ -99,13 +118,14 @@ private extension ASCFiltersViewController {
...
-            collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
-            collectionView.bottomAnchor.constraint(equalTo: showResultsButton.topAnchor),
-        ])
+        collectionView.anchor(
+            top: view.safeAreaLayoutGuide.topAnchor,
+            left: view.leftAnchor,
+            bottom: view.bottomAnchor,
+            right: view.rightAnchor,
+            leftConstant: 16,
+            rightConstant: 16
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:127]: Use own extensions to layout

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 8]
```diff
@@ -121,10 +141,19 @@ private extension ASCFiltersViewController {
...
         dismiss(animated: true)
     }
+
+    @objc func onShowResultsButtonTapped() {
+        print("call onShowResultsButtonTapped")
+    }
 }
 
 extension ASCFiltersViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:151]: Need to rewrite the code below. It is necessary to control the appearance at the view level and not at the controller level. Change view states through properties. Check the appearance on the light and dark theme of the system.

File: Documents/Documents/Code/Controllers/Filters/View/ASCFiltersCollectionViewCell.swift [hunk 9]
```diff
@@ -9,7 +9,8 @@
 import UIKit
 
 class ASCFiltersCollectionViewCell: UICollectionViewCell {
-    static let identifier = "ASCFiltersCollectionViewCell"
+    static let identifier = String(describing: ASCFiltersCollectionViewCell.self)
```
Comment (ayuzhin) [inline Documents/Documents/Code/Controllers/Filters/View/ASCFiltersCollectionViewCell.swift:12]: Remove hard code

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift (hunk 8 above)
Comment (ayuzhin) [reply Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:151]: Need to fix a **bug** while scrolling

File: Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift [hunk 10]
```diff
@@ -9,72 +9,92 @@
...
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Documents", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Presentations", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Spreadsheets", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Images", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Media", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("Archives", comment: ""), isSelected: false),
+                ASCDocumentsFilterModel(filterName: NSLocalizedString("All files", comment: ""), isSelected: false),
+            ]
+        ),
+        ASCDocumentsSectionModel(
```
Comment (chernyshevpavel) [inline Documents/Documents/Code/Controllers/Filters/ASCFiltersViewController.swift:28]: Make a protocol with one method - return ASCDocumentsFilterModel list
Then the class that will implement it. For example ASCDocumentsFixedFilterModelsProvider.
The current class should receive this provider as a service in the constructor or in a property.
They can be divided into sections by an additional attribute distinguishing them by groups

**Pull request merged**